### ISSUE TITLE
CLI proposal: Rename `develop` into `install`, remove `develop`

### ISSUE THREAD

**Issue by moigagoo** [sdispater/poetry#181]
With the introduction of `poetry develop`, `poetry install` seems redundant. Both commands install the deps, but `develop` also makes the package importable from any script executed with `poetry run`. It seems `install` is just a worse version of `develop` now.

Making this change will simplify the CLI, reduce confusion (`install` command would install the package), all without breaking the current `poetry install` behavior.

**Comment by ambv** [comment]
I agree this would be cleaner.

**Comment by jacebrowning** [comment]
I also have a use case for `poetry develop --no-dev`. I want to be able to run a script on my build server that imports my package, but I don't want to waste time installing all the development dependencies.

Merging `develop` into `install` (and keeping `--no-dev`) would meet this requirement.

**Comment by jacebrowning** [comment]
I'm not too familiar with the internals.

Does the package have to be editable in order to import it from scripts? Are `install` and `develop` equivalent to the `python setup.py <...>` commands? Does the existence of `*.egg-info` necessarily mean the package was installed in an editable manner?

**Comment by dfee** [comment]
I really don't understand the point of `develop`. Maybe 10 days ago the situation was different. But in my last 5 days of using this evolving tool, I cannot understand how `poetry install` is not a superset of `poetry develop`. What would be lost in straight up cutting `poetry develop`?

Next: `poetry add` vs `poetry install`.
The latter command – `poetry install` – installs *the current directory*. Whereas the former command – `poetry add` – adds a requirement to the `pyproject.toml` and then **installs** that package. See the confusion? The term **install** is overloaded.

But it kinda gets worse. If you want to **install** a package in development mode, you need to run two commands: `poetry add --path ../mypkg mypkg` and then `poetry install --develop mypkg`.

To that end, I think `poetry add` **should not** install a package, or it should allow the `--develop` option.

**Comment by jacebrowning** [comment]
@sdispater could you weight in on this proposal when you have time?

**Comment by cauebs** [comment]
I guess this can be closed now. 
I'm not very happy with how the maintainer basically ignored this discussion. I get that it's _his_ project, but we're all here trying to improve it. It would be nice to just get some sign that actively thinking about these problems and stating our opinions isn't utterly pointless.

**Comment by sdispater** [comment]
@cauebs I'm sorry you feel that way. That was not my intention. Be assured that I value every feedback and I try to be as responsive as possible. So even though I do not necessarily respond it does not mean that I am not taking into account the feedback people are giving.

Specifically for this issue, the introduction of `develop` was a poorly though out idea and this issue made me aware of it. However, more important issues required my intention and, in the end, I put it on the backlog but I kept it in the back of my mind. And once the bulk of the work for release `0.12.0` was done I found the time to address this without notifying the participants and telling them that this concern had been heard and that it was being taken care of.

This is poor communication on my part, I agree, but rest assured that I welcome any discussion that can lead to improving Poetry.
